frontend/javascripts/oxalis/model/reducers/annotation_reducer.js

```diff
@@ -1,15 +1,34 @@
 // @flow
 
 import update from "immutability-helper";
+import _ from "lodash";
 
 import type { Action } from "oxalis/model/actions/actions";
-import type { OxalisState } from "oxalis/store";
+import type { OxalisState, UserBoundingBox } from "oxalis/store";
 import { type StateShape1, updateKey, updateKey2 } from "oxalis/model/helpers/deep_update";
 import { convertServerAnnotationToFrontendAnnotation } from "oxalis/model/reducers/reducer_helpers";
 
 const updateTracing = (state: OxalisState, shape: StateShape1<"tracing">): OxalisState =>
   updateKey(state, "tracing", shape);
 
+const updateUserBoundingBoxes = (state: OxalisState, userBoundingBoxes: Array<UserBoundingBox>) => {
+  const updaterObject = {
+    userBoundingBoxes: {
+      $set: userBoundingBoxes,
+    },
+  };
+  const maybeSkeletonUpdater = state.tracing.skeleton ? { skeleton: updaterObject } : {};
+  const maybeVolumeUpdater = state.tracing.volume ? { volume: updaterObject } : {};
+  const maybeReadOnlyUpdater = state.tracing.readOnly ? { readOnly: updaterObject } : {};
+  return update(state, {
```

Comment:
Could you please add a comment somewhere here that bounding boxes are always mirrored to all present tracing objects.